### PR TITLE
(DPS-2739) Migrate SideNav component 

### PR DESCRIPTION
**Pull request by arturfracala**
## 🚀 Links

- [Feature branch](https://developers.labs.livechat.com/docs/feature/DPS-2739)
- [Jira](https://livechatinc.atlassian.net/browse/DPS-2739)

## 📓 Description

This is the initial version of migrated SideNav. Remember that there is still work to be done here. 

Missing steps: 

- [ ] Add buildArticle script to build command
- [x] Add support for h4 headings 
- [x] Add missing slugs to nodes in headingsOffsetMap (crucial for active paths)
- [x] Investigate why isExtended var doesn't work
- [x] Resolve versioning problem
- [x] Check working of the script that builds arrays of Articles 

Screenshot 2021-04-22 at 16 59 27

## ✅ Checklist (for API docs)

- Changelog
- API versions
- Web & RTM
- **Table:** Available methods/webhooks/pushes
- **Table:** Specifics

For more guidelines, see [Updating API docs](https://livechatinc.atlassian.net/wiki/spaces/PAT/pages/761529400/Updating+API+docs).

### Content

- New content
- Proofreading/content fixes
  
### Features (for the website)

- New feature
- Bug fix
- Breaking change

### CODE REVIEW

File: scripts/buildArticles.js [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+fs.readFile("../src/configs/categories.json", (err, data) => {
+  if (err) throw err;
+  const categories = JSON.parse(data);
+
+  const getDirectories = (src, callback) => glob(src + "/**/*", callback);
+
+  const articles = [];
+
+  categories.forEach((element) => {
```
Comment (jakubsikora) [inline scripts/buildArticles.js:16]: Maybe `category` ?

```suggestion
  categories.forEach((category) => {
```

File: scripts/buildArticles.js [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+      article["title"] = data["title"];
+    } else {
+      article["title"] = element["slug"];
+    }
+
+    articleGroupedByCategory.push(article);
+
+    element.items.forEach((item) => {
+      const article = {};
```
Comment (jakubsikora) [inline scripts/buildArticles.js:46]: Can we create a function which is formatting article so you could use it here and also in line 20. I know there are some differences there, but you should be able to make some generic.

**Review by jakubsikora — CHANGES_REQUESTED** [review]
Wow, this looks complex, good work! I have few comments.

**Review by dominikslawkowski — APPROVED** [review]
gj

**Review by jakubsikora — CHANGES_REQUESTED** [review]
Just one comment. The issue with memory leak will be investigated after we merge this.

**Review by jakubsikora — APPROVED** [review]

**Pull request merged**